### PR TITLE
Use Ruff for linting

### PR DESCRIPTION
**Pull request by stchris**
https://github.com/charliermarsh/ruff replaces flake8 + black (and can also auto-fix lots of things).

```
$ time make lint
ruff check .
make lint  0.01s user 0.03s system 47% cpu 0.091 total
```

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -35,11 +35,8 @@ jobs:
       - name: Install development dependencies
         run: make dev
 
-      - name: Check code formatting
-        run: make format-check
-
-      - name: Run linter
-        run: make lint
+      - name: Run linter (ruff)
+        uses: chartboost/ruff-action@v1
```
Comment (tillprochaska) [inline .github/workflows/build.yml:39]: What is the advantage of using this action over just `make lint` (which will now call ruff, too)?

File: Makefile [hunk 2]
```diff
@@ -27,20 +27,14 @@ test-ui:
...
 lint:
-	flake8 aleph/
+	ruff check .
 
 lint-ui:
 	$(UIDOCKER) npm run lint
 
-format:
-	black aleph/
-
```
Comment (tillprochaska) [inline Makefile:37]: Hm, but black doesn’t replace a formatter, right?

> As a project, Ruff is designed to be used alongside Black and, as such, will defer implementing stylistic lint rules that are obviated by autoformatting.

https://beta.ruff.rs/docs/faq/#is-ruff-compatible-with-black

**Review by tillprochaska — APPROVED** [review]
Left two comments/questions, but in general, I like ruff and am fine with whatever tooling everyone else prefers :)

File: .github/workflows/build.yml (hunk 1 above)
Comment (stchris) [reply .github/workflows/build.yml:39]: None, I guess. I will change it to be consistent, making sure we use the same version we have pinned.

File: Makefile (hunk 2 above)
Comment (stchris) [reply Makefile:37]: You are totally right, my bad. Reverting this.

**Comment by stchris** [comment]
Thanks for your careful review and great comments. I've made the changes and adapted the corresponding `ingest-file` PR as well. 👍

**Pull request merged**